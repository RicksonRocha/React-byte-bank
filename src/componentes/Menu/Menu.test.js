import { render, screen } from '@testing-library/react'
import Menu from './index'

test('Render home page link', () => {
    render(<Menu/>);
    const linkHomepage = screen.getByText('Inicial')
    expect(linkHomepage).toBeInTheDocument()
})

test('Render link list', () => {
    render(<Menu/>);
    const linkList = screen.getAllByRole('link')
    expect(linkList).toHaveLength(4)
})

test('does not render link to extract', () => {
    render(<Menu/>);
    const linkExtract = screen.queryByText('Extrato')
    expect(linkExtract).not.toBeInTheDocument()
})

test('Render a list link with link class', () => {
    render(<Menu/>);
    const links = screen.getAllByRole('link')
    links.forEach(link => expect(link).toHaveClass('link'))
    expect(links).toMatchSnapshot()
})