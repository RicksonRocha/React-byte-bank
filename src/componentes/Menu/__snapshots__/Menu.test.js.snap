// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a list link with link class 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
